packages/broker-service/Dockerfile

```diff
@@ -21,15 +21,42 @@ RUN --mount=type=secret,id=npm,target=/root/.npmrc yarn install --frozen-lockfil
 
 USER appuser
 
-# Install dependencies with a frozen lockfile
-# RUN yarn install --frozen-lockfile
+USER root
+
+# Install dependencies with a frozen lockfile and secret npmrc
+RUN --mount=type=secret,id=npm,target=/root/.npmrc yarn install --frozen-lockfile
+
+USER appuser
+
 
 # Build the necessary workspaces
 RUN yarn workspace @zkdb/storage build && \
     yarn workspace @zkdb/broker-service build
 
+# Use our customized image based on Node.js 20 slim image as a runner
+FROM orochinetwork/node:20-alpine AS runner
+
+# Create a non-root user and group in the runner stage
+RUN addgroup -S appuser && adduser -S -G appuser -h /usr/src/app appuser
+
+# Set the working directory
+WORKDIR /usr/src/app
+
+# Copy the necessary directories and node_modules from the builder stage
+COPY --from=builder /usr/src/app/packages/storage /usr/src/app/packages/storage
```

Comment:
Does it need change ownership?

Reply:
I think it doesn't. Because after copying all of neccesary file, I run
`RUN chown -R appuser:appuser /usr/src/app`, which ensures that the working directory is owned by the user we created. I use root to copy to make sure that all files are copied correctly.